### PR TITLE
cicd pipeline and image

### PR DESCRIPTION
**Pull request by whoisxx**
## Proposed changes

- Apply CICD pipeline process(rc deploy without createrepo, change repo EC2 to S3, parallel packaging.. etc)
- Add Dockerfile for rpmbuild image(install awscli, createrepo / alpine --> centos7)

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [x] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,21 +1,216 @@
...
-  rpmbuild:
+  rpm-executor:
+    working_directory: /go/src/github.com/klaytn/klaytn
     docker:
-      - image: klaytndevops/circleci-rpmbuild:1.13.3
+      - image: klaytn/circleci-rpmbuild:1.13.3-4
+  default:
+    working_directory: /go/src/github.com/klaytn/klaytn
+    docker:
+      - image: circleci/golang:latest
```
Comment (jeonghonah) [inline .circleci/config.yml]: Is it okay to use golang:latest always by default?

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,21 +1,216 @@
...
+          name: "rpm tagging"
+          command: make rpm-all
+      - run: 
+          name: "upload S3 repo"
+          command: |
+            PLATFORM_SUFFIX=$(uname -s | tr '[:upper:]' '[:lower:]')-$(uname -m)
+            for item in kcn kpn ken kscn kspn ksen kbn kgen homi; 
+            do
+              TARGET_RPM=$(find $item-linux-x86_64/rpmbuild/RPMS/x86_64/ | awk -v pat="$item(d)?-v" '$0~pat')
+              aws s3 cp $TARGET_RPM s3://$FRONTEND_BUCKET/packages/rhel/7/prod/
```
Comment (jeonghonah) [inline .circleci/config.yml]: Question:
Is rpm is fixed to rhel 7?

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,21 +1,216 @@
...
+      - run:
+          name: "rpm tagging baobab"
+          command: |
+              make rpm-baobab-kcn
+              make rpm-baobab-kpn
+              make rpm-baobab-ken
+      - run: 
+          name: "upload S3 repo"
+          command: |
+            for item in kcn kpn ken;
```
Comment (jeonghonah) [inline .circleci/config.yml]: kscn, kspn, ksen might be missing.

File: .circleci/config.yml [hunk 4]
```diff
@@ -26,50 +221,66 @@ jobs:
...
           name: "Run test others"
-          command: |
-            make test-others
+          command: make test-others
+  
+  pass-tests:
+    executor: default
+    steps:
+      - run: 
+          name: "tests Success!"
```
Comment (jeonghonah) [inline .circleci/config.yml]: Minor comment:
```suggestion
          name: "tests pass!"
```

File: .circleci/config.yml [hunk 5]
```diff
@@ -79,18 +290,20 @@ jobs:
           command: |
             set -e
             export GOPATH=/go
-            git clone --depth 1 https://$TEST_TOKEN@github.com/klaytn/klaytn-tests.git tests/testdata
-            make cover
+            git clone --depth 1 https://$TEST_TOKEN@github.com/klaytn/klaytn-tests.git tests/testdata   
+            make cover
```
Comment (jeonghonah) [inline .circleci/config.yml]:
```suggestion
            make cover
```

File: .circleci/config.yml [hunk 6]
```diff
@@ -79,18 +290,20 @@ jobs:
           command: |
             set -e
             export GOPATH=/go
-            git clone --depth 1 https://$TEST_TOKEN@github.com/klaytn/klaytn-tests.git tests/testdata
-            make cover
+            git clone --depth 1 https://$TEST_TOKEN@github.com/klaytn/klaytn-tests.git tests/testdata
```
Comment (jeonghonah) [inline .circleci/config.yml]:
```suggestion
            git clone --depth 1 https://$TEST_TOKEN@github.com/klaytn/klaytn-tests.git tests/testdata
```

File: .circleci/config.yml [hunk 7]
```diff
@@ -99,409 +312,206 @@ jobs:
...
-              KLAYTN_PACKAGE_NAME="${item}-baobab-${KLAYTN_VERSION}-${KLAYTN_RELEASE_NUM}-linux-amd64.tar.gz"
-              tar czf $KLAYTN_PACKAGE_NAME $PACK_NAME
-              aws s3 cp $KLAYTN_PACKAGE_NAME s3://klaytn-ops-stuff/releases/klaytn/$KLAYTN_VERSION/
-              aws s3 cp $KLAYTN_PACKAGE_NAME s3://klaytn-ops-stuff/releases/klaytn/latest/
-            done
+      - pre-build
+      - build-packaging:
+          os-network: "tar-baobab-linux-amd64-all"
+      - upload-repo:
+          item: "kcn kpn ken"
```
Comment (jeonghonah) [inline .circleci/config.yml]: Please consider kscn, kspn, ksen and homi.

File: .circleci/config.yml [hunk 8]
```diff
@@ -99,409 +312,206 @@ jobs:
...
-          name: "Install awscli"
-          command: |
-            curl -fsSL https://bootstrap.pypa.io/get-pip.py -o get-pip.py
-            python get-pip.py --user
-            ~/.local/bin/pip install awscli --upgrade --user
+      - pre-build
+      - build-packaging:
+          os-network: "tar-baobab-darwin-amd64-all"
+      - upload-repo:
+          item: "kcn kpn ken"
```
Comment (jeonghonah) [inline .circleci/config.yml]: Same as above.

File: .circleci/config.yml [hunk 9]
```diff
@@ -99,409 +312,206 @@ jobs:
...
             - rpm-tagged
             - rpm-tagged-baobab
+            - packaging-linux
+            - packaging-linux-baobab
+            - packaging-darwin
+            - packaging-darwin-baobab
           filters:
             tags:
-              only: /^v.*/
+              only: /^v[0-9]+\.[0-9]+\.[0-9]+/
```
Comment (jeonghonah) [inline .circleci/config.yml]:
```suggestion
              only: /^v[0-9]+\.[0-9]+\.[0-9]+.*/
```

File: .circleci/config.yml [hunk 10]
```diff
@@ -99,409 +312,206 @@ jobs:
...
+      - tagger-verify:
+         filters:
             tags:
-              only: /.*/
-
-      - test-others:
-          filters:
-            tags:
-              only: /.*/
+              only: /^v[0-9]+\.[0-9]+\.[0-9]+/
```
Comment (jeonghonah) [inline .circleci/config.yml]:
```suggestion
              only: /^v[0-9]+\.[0-9]+\.[0-9]+.*/
```

File: .circleci/config.yml [hunk 11]
```diff
@@ -99,409 +312,206 @@ jobs:
...
+            - rpm-tagged
+            - rpm-tagged-baobab
+            - packaging-linux
+            - packaging-linux-baobab
+            - packaging-darwin
+            - packaging-darwin-baobab
+          filters: 
             tags:
-              only: /^v.*/
+              only: /^v[0-9]+\.[0-9]+\.[0-9]+-rc.*/
```
Comment (jeonghonah) [inline .circleci/config.yml]:
```suggestion
              only: /^v[0-9]+\.[0-9]+\.[0-9]+-rc\.[0-9]+.*/
```

File: build/Dockerfile-go1.13.3-4-rpmbuild [hunk 12]
```diff
@@ -0,0 +1,7 @@
+FROM centos:centos7
+
+RUN curl https://dl.google.com/go/go1.13.linux-amd64.tar.gz | tar xzvf - -C /usr/local
```
Comment (jeonghonah) [inline build/Dockerfile-go1.13.3-4-rpmbuild]:
```suggestion
RUN curl https://dl.google.com/go/go1.13.4.linux-amd64.tar.gz | tar xzvf - -C /usr/local
```

File: .circleci/config.yml (hunk 1 above)
Comment (whoisxx) [reply .circleci/config.yml]: this image will run for tagging, open PR, echo.. etc. do nothing about build/package.
but, when it makes git tag, golang will used to check verison from version.go. 
If checking version by version.go will highly affected by go version, I can specify version number

**Pull request merged**